apps/bestofjs-nextjs/src/app/backend-search-requests.tsx

```diff
@@ -14,7 +14,9 @@ export function getHotProjectsRequest(count = 5) {
 
 export function getLatestProjects() {
   return {
-    sort: {}, // TODO sort by `addedAt` when it's available in data instead of relying on the default sort
+    sort: {
+      added_at: -1,
```

Comment:
I'm not understanding this line.

I'd like to spend more time to dig into this but I cannot.

So I will start the discussion around my understanding as a "new dev" here.


You encapsulate the queries in `backend-search-requests.tsx` (Not .ts?)

These are basically configuration objects that will be consumed by your ORM right?

I'm just a bit worried that the values are not self explanatory, something like `limit: 5` is easy enough to understand without having to find some extra documentation, but for now the sort configuration looks like black magic to me.

Reply:
What we call backend in the Next.js app is the big JSON file that contains aggregated data (including the trends), rebuilt every day. The Postgres DB is used only when opening the project "details" page, not to generate lists of projects.

We query this file using MongoDB-format queries (using Mingo package) when showing any list of projects on the screen.

I chose this solution because MongDB queries are just plain JS objects that are very convenient to express the criteria, the sort order, the number of projects we want.

So this file `backend-search-request` only contains two queries that I used in different places, to keep the code dry. It should be `.ts`, not `.tsx` 👍 

The following means: the MongoDB query to get the 5 first projects by `added_at` date.

```ts
export function getLatestProjects() {
  return {
    sort: {
      added_at: -1,
    },
    limit: 5,
  };
}
```